src/Control/Concurrent/Extra.hs

```diff
@@ -152,14 +157,34 @@ readVar (Var x) = readMVar x
 writeVar :: Var a -> a -> IO ()
 writeVar v x = modifyVar_ v $ const $ pure x
 
+-- | Strict variant of 'writeVar'
+writeVar' :: Var a -> a -> IO ()
+writeVar' v x = modifyVar_' v $ const $ pure x
+
 -- | Modify a 'Var' producing a new value and a return result.
 modifyVar :: Var a -> (a -> IO (a, b)) -> IO b
 modifyVar (Var x) f = modifyMVar x f
 
+-- | Strict variant of 'modifyVar''
+modifyVar' :: Var a -> (a -> IO (a, b)) -> IO b
+modifyVar' (Var x) f = do
+    (newContents, res) <- modifyMVar x $ \v -> do
+        (newContents, res) <- f v
+        return (newContents, (newContents, res))
```

Comment:
For this repo, I have a lint on to say use `pure` instead of `return`, for consistency.